### DISCUSSION TITLE
Disable remote caching on vercel

### DISCUSSION THREAD

**Discussion by PabloSzx** [vercel/turborepo#671] [Help] [answered]
How can I disable remote caching on vercel? in a project using prisma it only gives issues, because it thinks that it cached when it didn't and nothing of prisma is correctly cached, and instead, nothing is generated and the build fails

---

I managed to fix the issue disabling cache on the specific repo that has the prisma instance:

```json
      "db#generate": {
        "cache": false
      },
```

but remote caching gives issues on projects with a lot of environment variables, where https://turborepo.org/docs/features/caching#alter-caching-based-on-environment-variables-and-files is not straight-forward and we have to add "--force" for the Next.js builds

**Comment by ScreamZ** [comment]
`npx turbo unlink` seems to be the solution

**Comment by jacksonblankenship** [comment]
I'm having this issue w/ Prisma and it's insanely annoying. I have to opt-out of remote caching just so my client will generate.

**Comment by tknickman** [comment, marked as answer]
You have a few options for disabling remote caching depending on your execution context. 

# Locally 

### Option 1. 
Run [turbo unlink](https://turborepo.org/docs/reference/command-line-reference#turbo-unlink)
> This will unlink the current repo from the Vercel remote cache. No runs will use the cache unless the repo is relinked

### Option 2
Run with [--force](https://turborepo.org/docs/reference/command-line-reference#--force)
> This will bypass the cache for any run that includes this argument. 

# On Vercel

### Option 1
[Disable globally](https://vercel.com/docs/concepts/monorepos/remote-caching)
> The Vercel Remote Cache can be disabled on an account level using the toggle found in account/billing ([see the docs for more details](https://vercel.com/docs/concepts/monorepos/remote-caching))

### Option 2
Update your build script to include [--force](https://turborepo.org/docs/reference/command-line-reference#--force)
> This will bypass the cache for any run that includes this argument. 

# In CI

### Option 1
Update your build script to include [--force](https://turborepo.org/docs/reference/command-line-reference#--force)
> This will bypass the cache for any run that includes this argument. 

---



With that being said, the Prisma issue you're experiencing sounds like it should be investigated separately - would you mind creating an issue for this with some more details?
